### ISSUE TITLE
VM with host only network does not start (virtualbox)

### ISSUE THREAD

**Issue by carlosefr** [hashicorp/vagrant#11476]
### Vagrant version

2.2.7

### Host operating system

macOS 10.14.6 (Mojave)

### Guest operating system

CentOS Linux 7.7

### Vagrantfile

```ruby
Vagrant.configure(2) do |config|
    config.vm.box = "carlosefr/centos-7"

    config.vm.provider "virtualbox" do |vm, override|
        override.vm.network "private_network", type: "dhcp"
    end
end
```

### Debug output

https://gist.github.com/carlosefr/3b698d0d3eb66a79d256365d93631665

### Expected behavior

The VM should start after issuing an `vagrant up` command.

### Actual behavior

The VM does not start, failing with the following error:
```
$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'carlosefr/centos-7'...
==> default: Matching MAC address for NAT networking...
==> default: Setting the name of the VM: vm-centos-test_default_1585577268195_4925
==> default: Clearing any previously set network interfaces...
A host only network interface you're attempting to configure via DHCP
already has a conflicting host only adapter with DHCP enabled. The
DHCP on this adapter is incompatible with the DHCP settings. Two
host only network interfaces are not allowed to overlap, and each
host only network interface can have only one DHCP server. Please
reconfigure your host only network or remove the virtual machine
using the other host only network.
```

Removing the DHCP server associated with `vboxnet0` allows the VM to come up, but **only once**:
```
VBoxManage dhcpserver remove --netname HostInterfaceNetworking-vboxnet0
```

After a `vagrant halt`, the VM won't come up again, failing with the same error as before.

This started happening after upgrading to **VirtualBox 6.1.4** (clean installation). It didn't happen with VirtualBox 6.0.18 (and doesn't happen with this version after downgrading).

### Steps to reproduce

1. `vagrant up` (error)
2. `VBoxManage dhcpserver remove --netname HostInterfaceNetworking-vboxnet0`
3. `vagrant up` (OK)
4. `vagrant halt`
5. `vagrant up` (error)
6. `VBoxManage dhcpserver remove --netname HostInterfaceNetworking-vboxnet0`
7. `vagrant up` (OK)
8. ...

**Comment by jbonhag** [comment]
Hi @carlosefr,

Thanks for reporting this issue. This was caused by a change in the way VirtualBox 6.1 reports DHCP servers, and is fixed by #11404.

Cheers and have a great day! 😄

**Comment by jbonhag** [comment]
See also #11403.
